utils/src/activityLog/activities/ExecuteActivity.ts

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { TreeItemCollapsibleState } from "vscode";
+import * as types from '../../../index';
+import { AppResource } from "../../../unified";
+import { AzExtParentTreeItem } from "../../tree/AzExtParentTreeItem";
+import { GenericTreeItem } from "../../tree/GenericTreeItem";
+import { nonNullProp } from "../../utils/nonNull";
+import { ActivityBase, ActivityTreeItemOptions } from "../Activity";
+
+
+interface ExecuteActivityData<C extends types.IActionContext> {
+    title: string;
+    context: C;
+}
+
+export class ExecuteActivity<C extends types.IActionContext> extends ActivityBase {
+
+    public constructor(private readonly data: ExecuteActivityData<C>, task: types.ActivityTask) {
+        super(task);
+    }
+
+    public initialState(): ActivityTreeItemOptions {
+        return {
+            label: this.data.title,
+            collapsibleState: TreeItemCollapsibleState.None,
+        }
+    }
+
+    public successState(): ActivityTreeItemOptions {
+        return {
+            label: this.labelOnDone,
```

Comment:
I'm not sure I totally understand what the difference between `this.data.title` and `this.labelOnDone `is.  Is it in case `data.title` changes at some point?

Reply:
I'm splitting the AzureWizard changes into a new PR, I'll respond to this feedback when I do that.